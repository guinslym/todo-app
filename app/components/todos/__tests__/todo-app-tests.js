import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { shallow } from 'enzyme';
import { MemoryRouter as Router } from 'react-router-dom';
import TodoApp from '../todo-app';

const mockStore = configureStore();

describe('<TodoApp/>', () => {
  let component;

  beforeEach(() => component = shallow(<TodoApp/>));

  it('should render', () => {
    expect(component.length).toBeTruthy();
  });

});

describe('<TodoApp/> snapshot', () => {

  it('should render correctly', () => {
    let store = mockStore({ todos: [] });
    const tree = renderer.create(
      <Provider store={store}><Router><TodoApp/></Router></Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

});
