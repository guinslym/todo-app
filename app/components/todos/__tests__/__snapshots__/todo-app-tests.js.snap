// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoApp/> snapshot should render correctly 1`] = `
<section>
  <h1>
    Todo App
  </h1>
  <section>
    <form
      onSubmit={[Function]}
    >
      <input />
      <button
        type="submit"
      >
        Add todo
      </button>
    </form>
  </section>
  <ul />
  <p>
    Show:
     
    <a
      className="active"
      href="/"
      onClick={[Function]}
      style={
        Object {
          "color": "black",
          "textDecoration": "none",
        }
      }
    >
      All
    </a>
     | 
    <a
      className={undefined}
      href="/active"
      onClick={[Function]}
      style={undefined}
    >
      Active
    </a>
     | 
    <a
      className={undefined}
      href="/completed"
      onClick={[Function]}
      style={undefined}
    >
      Completed
    </a>
  </p>
</section>
`;
